src/Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -665,12 +665,12 @@ private void CheckStake(Block block, ChainedBlock chainTip)
 
                     // We have to make sure that we have no future timestamps in
                     // our transactions set.
-                    foreach (Transaction transaction in block.Transactions)
+                    for (int i = block.Transactions.Count - 1; i >= 0; i--)
```

Comment:
Hmm I would have done it differently but this is also good

Reply:
You mean using a 2nd list like `toBeRemoved` or list.RemoveAll(x => x.timestamp > ...)? 

First solution will consume more memory and 2nd one will force us to get rid of the trace log. 
Or you had a different one in mind?

Reply:
Yeah using as second list, it wont consume more memory as you are not creating new instances just moving them to a new list. but this is good as well no problem.

Reply:
Ok, cool. 
Though I don't mind changing it if you'd like.